easybuild/easyblocks/p/pytorch.py

```diff
@@ -42,6 +42,98 @@
 from easybuild.tools.systemtools import POWER, get_cpu_architecture
 
 
+def get_count_for_pattern(regex, text):
+    """Match the regexp containing a single group and return the integer value of the matched group.
+       Return zero if no or more than 1 match was found and warn for the latter case
+    """
+    match = re.findall(regex, text)
+    if len(match) == 1:
+        return int(match[0])
+    elif len(match) > 1:
+        # Shouldn't happen, but means something went wrong with the regular expressions.
+        # Throw warning, as the build might be fine, no need to error on this.
+        warn_msg = "Error in counting the number of test failures in the output of the PyTorch test suite.\n"
+        warn_msg += "Please check the EasyBuild log to verify the number of failures (if any) was acceptable."
+        print_warning(warn_msg)
+    return 0
+
+
+def extract_failed_tests_info(tests_out):
+    """
+    Extract information about failed tests from output produced by PyTorch tests
+    """
+
+    # Create clear summary report
+    failure_report = ""
+    failure_cnt = 0
+    error_cnt = 0
+    failed_test_suites = []
+
+    # Look for patterns like:
+    # Ran 219 tests in 67.325s
+    #
+    # FAILED (errors=10, skipped=190, expected failures=6)
+    # test_fx failed!
+    regex = (r"^Ran (?P<test_cnt>[0-9]+) tests.*$\n\n"
+             r"FAILED \((?P<failure_summary>.*)\)$\n"
+             r"(?:^(?:(?!failed!).)*$\n)*"
+             r"(?P<failed_test_suite_name>.*) failed!(?: Received signal: \w+)?\s*$")
+
+    for match in re.finditer(regex, tests_out, re.M):
+        # E.g. 'failures=3, errors=10, skipped=190, expected failures=6'
+        info = match.groupdict()
+        failure_summary = info['failure_summary']
+        test_cnt, test_suite = info['test_cnt'], info['failed_test_suite_name']
```

Comment:
What was wrong with the original, shorter:
```suggestion
        failure_summary = m.group('failure_summary')
        total, test_suite = m.group('test_cnt', 'failed_test_suite_name')
```

Reply:
I mainly started changing this due to the single-letter variable, I don't have a good reason to favor `.groupdict` over using `.group` multiple times

Reply:
The latter has the advantage that getting multiple values at once is easy :)